### ISSUE TITLE
Map dual carriageways and improve inferences

### ISSUE THREAD

**Issue by dabreegster** [dabreegster/abstreet#230]
Sidewalks and parking lanes probably don't exist on the inner part of dual carriageways. Seattle doesn't have `dual_carriageway=yes`, but it'd be easy for me to go map it. Buried in the parking mapper is a tool that detects these. (I don't trust this tool enough to use it to automatically infer dual carriageways; I want to manually review its results.)

**Comment by matkoniecz** [comment]
That would be probably a good moment to document this tag, including explaining why it is useful and discussing some edge cases.

**Comment by dabreegster** [comment]
Definitely. For starters, I notice https://wiki.openstreetmap.org/wiki/Key:dual_carriageway and https://wiki.openstreetmap.org/wiki/Tag:dual_carriageway%3Dyes are almost duplicates

**Comment by matkoniecz** [comment]
Oh, I forgot that this pages are already created - I wanted to create them. I soft-redirected https://wiki.openstreetmap.org/wiki/Key:dual_carriageway

**Comment by matkoniecz** [comment]
I added https://wiki.openstreetmap.org/wiki/Tag:dual_carriageway%3Dyes#Rationale

**Comment by dabreegster** [comment]
https://www.redblobgames.com/articles/curved-paths/cities-in-motion-2-full.jpg gave me a thought -- filling in the visual gaps between divided highways that aren't a bridge or tunnel could be quite an improvement, and additional motivation to tag this. Examples where it'd be helpful:
![Screenshot from 2020-11-12 10-48-34](https://user-images.githubusercontent.com/1664407/98982760-b5ff3480-24d4-11eb-833d-ec0882fbd069.png)
![Screenshot from 2020-11-12 10-47-57](https://user-images.githubusercontent.com/1664407/98982763-b7306180-24d4-11eb-983a-472dc576efe7.png)
And a case where maybe we should hold off or do something different, since there's a grassy median already mapped:
![Screenshot from 2020-11-12 10-49-09](https://user-images.githubusercontent.com/1664407/98982804-c6afaa80-24d4-11eb-832d-9145e82eafdf.png)

**Comment by dabreegster** [comment]
![Screenshot from 2020-11-12 14-06-19](https://user-images.githubusercontent.com/1664407/99002004-a55cb780-24f0-11eb-9490-5bdfde248704.png)
![Screenshot from 2020-11-12 14-06-04](https://user-images.githubusercontent.com/1664407/99002007-a68de480-24f0-11eb-8a39-fb81b4ae7614.png)
Got something started, but it happens on both edges of every one-way road. I might pull in some of the dual carriageway heuristics from the parking mapper, but the more straightforward thing would be to add this tag manually and look for it. :)

**Comment by dabreegster** [comment]
https://overpass-turbo.eu/s/102L -- woohoo, a bunch in Seattle mapped! Updating the code to only attempt walking-around-the-block near these

**Comment by 1ec5** [comment]
By the way, the median strip in a divided highway (dual carriageway) can also be represented as part off highway land use. There was [a proposal](https://wiki.openstreetmap.org/wiki/Proposed_features/landuse%3Dhighway) to map highway land use as `landuse=highway`, but the vote on it seems to have suffered from hasty conflation with other concepts like `area:highway` (which is for just the pavement). Some print maps shade highway land use to set off large junctions from the rest of the map; this treatment would benefit from `landuse=highway`:

<img src="https://user-images.githubusercontent.com/1231218/99360715-ded15200-2865-11eb-9d54-411333aa6ae5.png" width="300"> <img src="https://user-images.githubusercontent.com/1231218/99360725-e09b1580-2865-11eb-91ab-802f7a64519d.png" width="300">

That said, `dual_carriageway=yes` is more convenient for treatments at larger scales (lower zoom levels), where a divided highway is drawn abstractly as a double line with a fixed color in the middle, regardless of the shape and size of the median in reality.

**Comment by dabreegster** [comment]
More ideas for rendering: https://twitter.com/CarlinoDustin/status/1328177931261079553
